### PR TITLE
Adding support for Ubuntu

### PR DESCRIPTION
**Pull request by neelimamukiri**
Primary changes in the installer are in https://github.com/contiv/ansible/commit/0bde886996d1668287d730cbc2ca868bf7c16e44

The installer changes below are to copy the ansible inventory to the config folder instead of a folder inside the container, to make it easier for troubleshooting.

Most of the Vagrantfile changes are to workaround differences in the Ubuntu/Centos box - different ethernet interface names, user name and key semantics.

### CODE REVIEW

File: cluster/Vagrantfile [hunk 1]
```diff
@@ -29,8 +29,10 @@ def create_cfg_info(num_nodes, node_ips)
            else
              {}
            end
-    node['control'] = ENV['CONTIV_CONTROL_IF'] || 'eth1'
-    node['data'] = ENV['CONTIV_DATA_IF'] || 'eth2'
+    def_ctrl_if =  ENV['CONTIV_NODE_OS'] && ENV['CONTIV_NODE_OS'] == 'ubuntu' ? 'enp0s8' : 'eth1'
```
Comment (dseevr) [inline cluster/Vagrantfile]: `Hash#[]` returns `nil` when a key is not found so you can remove the first half of the boolean condition on checks like these without issue.

`nil == "any string"` is false (or more specifically `NilClass#==` just returns false if the input is a string)

File: cluster/Vagrantfile [hunk 2]
```diff
@@ -86,15 +98,25 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
...
+        s.args = [ENV['http_proxy'] || '', ENV['https_proxy'] || '']
       end
+      if c.vm.box != 'ubuntu/xenial64'
+        c.vm.provision 'shell' do |s|
+          s.inline = provision_net
+          etc_file = (ENV['VAGRANT_CWD'] || '.') + '/.etc_hosts'
+          hosts = File.read(etc_file)
+          s.args = [hosts]
+        end
+        end
```
Comment (dseevr) [inline cluster/Vagrantfile]: This line needs to be outdented two spaces

File: cluster/Vagrantfile [hunk 3]
```diff
@@ -86,15 +98,25 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
         # Download rhel7.2 box from https://access.redhat.com/downloads/content/293/ver=2/rhel---7/2.0.0/x86_64/product-software
         # Add it as rhel7 vagrant box add rhel-cdk-kubernetes-7.2-29.x86_64.vagrant-virtualbox.box --name=rhel7
         c.vm.box = 'rhel7'
+        config.ssh.insert_key = false
+      elsif ENV['CONTIV_NODE_OS'] && ENV['CONTIV_NODE_OS'] == 'ubuntu'
+        c.vm.box = 'ubuntu/xenial64'
```
Comment (dseevr) [inline cluster/Vagrantfile]: This string appears five times in this file... I'd consider making it a constant

File: cluster/Vagrantfile [hunk 4]
```diff
@@ -86,15 +98,25 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
         # Download rhel7.2 box from https://access.redhat.com/downloads/content/293/ver=2/rhel---7/2.0.0/x86_64/product-software
         # Add it as rhel7 vagrant box add rhel-cdk-kubernetes-7.2-29.x86_64.vagrant-virtualbox.box --name=rhel7
         c.vm.box = 'rhel7'
+        config.ssh.insert_key = false
+      elsif ENV['CONTIV_NODE_OS'] && ENV['CONTIV_NODE_OS'] == 'ubuntu'
```
Comment (dseevr) [inline cluster/Vagrantfile]: This can be reduced

File: cluster/Vagrantfile [hunk 5]
```diff
@@ -43,13 +45,24 @@ echo "export https_proxy='$2'" >> /etc/profile.d/envvar.sh
...
+sudo apt-get install python -y
+mkdir -p /home/ubuntu/.ssh/
+cat <<EOF >> /home/ubuntu/.ssh/authorized_keys
+$1
+EOF
+SCRIPT
+
 # begin execution here
 num_nodes = 2
 if ENV['CONTIV_NODES'] && ENV['CONTIV_NODES'] != ''
```
Comment (dseevr) [inline cluster/Vagrantfile]: This can be reduced

File: install/ansible/install.sh [hunk 6]
```diff
@@ -86,6 +87,22 @@ fi
...
+ansible all $ans_opts -i $host_inventory -m setup -a 'filter=ansible_distribution*' >& $inventory_log
+egrep 'FAIL|UNREACHABLE' $inventory_log >& /dev/null
+if [ $? -eq 0 ]; then
+   echo "WARNING"
+   echo "WARNING Some of the hosts are not accessible via passwordless SSH"
+   echo "WARNING"
+   echo "`egrep 'FAIL|UNREACHABLE' $inventory_log`"
+   echo " "
+   echo "This means either the host is unreachable or passwordless SSH is not"
+   echo "set up for it. It is RECOMMENDED that you resolve this before proceeding."
```
Comment (dseevr) [inline install/ansible/install.sh]: Is it really _recommended_ if the script exits with a 1 right afterwards instead of continuing? :D

**Comment by dseevr** [comment]
It looks like there's a log file being created under `cluster` and a `release` directory which gets filled with build artifacts... can these both be `.gitignore`d?

**Comment by neelimamukiri** [comment]
I added the files to .gitignore and made the rest of the reductions :)

**Pull request merged**